string msgType
string ctrlType
int32 value





